cassiopeia/tools/fitness_estimator/_lbi_jungle.py

```diff
@@ -0,0 +1,116 @@
+import os
+import sys
+import tempfile
+from typing import Optional
+
+import networkx as nx
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+print(f"dir_path = {dir_path}")
+sys.path.append(os.path.join(dir_path, "_jungle"))
+import jungle as jg
+import numpy as np
+
+from cassiopeia.data import CassiopeiaTree
+
+from ._FitnessEstimator import FitnessEstimator, FitnessEstimatorError
+
+
+def _to_newick(tree: nx.DiGraph, record_branch_lengths: bool = False) -> str:
```

Comment:
I think this might be a slight variation of something already in our codebase, but tailored to the Jungle format. Or it might be identical, we can check.

Reply:
I think this is the same as `cassiopeia.data.utilities.to_newick`

Reply:
It seems like they are slightly different, to account for internal node names. The original version in `cassiopeia.data.utilities.to_newick` has:
```
        if record_branch_lengths and g.in_degree(node) > 0:
            parent = list(g.predecessors(node))[0]
            weight_string = ":" + str(g[parent][node]["length"])
```
whereas the version I wrote here has two more lines:
```
        if record_branch_lengths and g.in_degree(node) > 0:
            parent = list(g.predecessors(node))[0]
            weight_string = ":" + str(g[parent][node]["length"])
            if not is_leaf:
                weight_string = node + weight_string
```
I suggest we just keep these two separate versions. Not worth trying to consolidate them. The one I wrote is a private function in a private module anyway.

Reply:
Got it, I agree it's fine to keep.